velox/common/caching/SsdFileTracker.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+#include <vector>
+
+namespace facebook::velox::cache {
+
+// Tracks reads on an SsdFile. Reads are counted for fixed size regions and
+// periodically decayd. Not thread safe, synchronization is the caller's
+// responsibility.
+class SsdFileTracker {
+ public:
+  void resize(int32_t size) {
+    regionScore_.resize(size);
+  }
+
+  void recordUse(int32_t region, int32_t bytes) {
+    regionScore_[region] += bytes;
+  }
+
+  void clearScore(int32_t region) {
+    regionScore_[region] = 0;
+  }
+
+  // Increments event count and periodically decays
+  // scores. 'totalEntries' is the count of distinct entries in the
+  // tracked file.
+  void newEvent(int32_t totalEntries);
```

Comment:
All other methods operate on regions, but this one is for the whole file, hence, would be nice to make that clear by using "file" in the method name. Perhaps, rename to fileAccessed or fileTouched. Event name is a bit too generic and doesn't match other methods. It would also be nice to group all region-specific methods together, e.g. move this method after regionFilled.